### DISCUSSION TITLE
vulture to help find and eliminate dead code?

### DISCUSSION THREAD

**Discussion by benwandrew** [AutoResearch/autora#417] [Ideas]
potentially only relevant down the line, but just wondering what folks think of using a tool like [vulture](https://github.com/jendrikseipp/vulture) as `autora` grows? [referenced](https://goodresearch.dev/tidy.html#delete-dead-code) in the good research codebook. could running something like this periodically be useful, or otherwise good practice? or could this introduce some liabilities, even when paired with unit tests for preserved functionality?

@hollandjg @gtdang, thoughts?

**Comment by hollandjg** [comment]
I like the look of that! It would be really interesting to run that on the core package, just to see if there's anything which has been missed by our existing checks. Hopefully, our packages will be slim enough that it'll be obvious from inspection that code is necessary or unused, but I'm not sure that's true. 

What we have already:
- Pycharm does a lot of this already, telling you which variables and imports you don't use (if you turn on warnings, or run "code inspection"). 
- In the past, I've used `coverage` which points out which bits of code you haven't tested at all, which is a good proxy for "unused" if you have extensive test cases.
